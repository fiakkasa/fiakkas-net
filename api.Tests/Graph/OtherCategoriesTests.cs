using HotChocolate.Execution;

namespace api.Tests;
public partial class SoftwareDevelopmentCategoriesTests
{
    public class OtherCategoriesTests(GraphFixture fixture) : IClassFixture<GraphFixture>
    {
        [Fact]
        public async Task OtherCategories_Should_Return_Data()
        {
            var executor = await fixture.GetRequestExecutor();

            var result = await executor.ExecuteAsync(
"""
{
  otherCategories {
    totalCount
    nodes {
      createdAt
      id
      internalId
      title
      updatedAt
      version
    }
  }
}
""");

            Func<IQueryResult> fn = result.ExpectQueryResult;
            fn.Should().NotThrow();
            fn().Errors.Should().BeNullOrEmpty();
            result.ToJson().MatchSnapshot();
        }

        [Fact]
        public async Task OtherCategoryById_Should_Return_Data()
        {
            var executor = await fixture.GetRequestExecutor();

            var result = await executor.ExecuteAsync(
"""
{
  otherCategoryById(id: "T3RoZXJDYXRlZ29yeQpnOWZkOTFmOGEyYTQ0NDUyMGE1YjE4YWExYzNjMjkxMzM=") {
    createdAt
    id
    internalId
    title
    updatedAt
    version
  }
}
""");

            Func<IQueryResult> fn = result.ExpectQueryResult;
            fn.Should().NotThrow();
            fn().Errors.Should().BeNullOrEmpty();
            result.ToJson().MatchSnapshot();
        }
    }
}
