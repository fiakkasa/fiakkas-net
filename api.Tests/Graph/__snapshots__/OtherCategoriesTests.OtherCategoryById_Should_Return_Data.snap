{
  "data": {
    "otherCategoryById": {
      "createdAt": "2024-01-01T00:00:00.000Z",
      "id": "T3RoZXJDYXRlZ29yeQpnOWZkOTFmOGEyYTQ0NDUyMGE1YjE4YWExYzNjMjkxMzM=",
      "internalId": "9fd91f8a-2a44-4520-a5b1-8aa1c3c29133",
      "title": "Title",
      "updatedAt": null,
      "version": 1
    }
  }
}
